### DISCUSSION TITLE
Sigma File Manager Next (v2.0) progress

### DISCUSSION THREAD

**Discussion by aleksey-hoffman** [aleksey-hoffman/sigma-file-manager#226] [Announcements]
> [!NOTE] 
> **Status**: v2 became the main branch and entered beta stage

## About 

I am currently re-writing Sigma File Manager using newer, better technologies (Rust, Vue 3, Typescript), as well as improving code quality. This will result in a faster, safer, more capable, and overall better app.

It comes out with v2.0 update.

For now, the new version will be called `Sigma File Manager Next (v2.0)`.

### Milestones
- [x] Migrate from Electron.js to a better framework - Tauri;
- [x] Make the app safer;
- [x] Reduce the size of the app installable file from `~150 MB` to `~50 MB`;
- [x] Reduce the size of the installed app from `~400 MB` to `~80 MB`;
- [x] Make the app faster:
  - [x] Faster app launch;
  - [x] Faster large directory loading;
  
### Base
- [x] Re-write the app frontend base with Vue 3, Typescript;
- [x] Re-write the app backend base with Rust;

### General
- [x] Localization;
- [x] Tray menu; 
- [x] New user settings system;
- [x] Window toolbar;
- [x] Workspaces;
- [x] Tabs;
- [x] File filter;
- [x] File search;
- [x] Window visual effects;
- [x] New file downloading system;
- [ ] Media file info extractor;

### Home view
- [x] Home view content section base;
- [x] Home banner section;
- [x] User directories section;
- [x] Drives section;

### Navigator view
- [x] Navigator view content section base;
- [x] **New feature:** split view;
- [x] **New feature:** ability to set custom size of directory elements;
- [x] Fix grid layout;
- [x] Improve styling and make it more consistent;
- [x] Action toolbar;
- [x] Navigator toolbar;
- [x] Sorting toolbar;
- [x] Info panel;

### Dashboard view
- [x] Dashboard view content section base;
- [x] Action toolbar;
- [x] Pinned section;
- [ ] Protected section;
- [x] Timeline section;

### Notes view
- [x] Will not be migrated from v1

### Settings view
- [x] Settings view content section base;
- [x] Navigation;
- [x] Settings component;

### Extensions view
- [ ] Extensions view content section base;
- [ ] Extension system for managing community extensions;

### Tabs
- [x] **New feature:** Split view support

**Comment by begin-theadventure** [comment]
Hooray!

**Comment by aleksey-hoffman** [comment]
### New feature | Split view
Youtube: https://www.youtube.com/watch?v=EdeaYASp-t8

![image](https://github.com/aleksey-hoffman/sigma-file-manager/assets/61761672/1699ced5-4e05-40cc-8f0e-4f0248e77604)

**Comment by gloritown** [comment]
Appreciate the Update! Do we have any timeline for the version 2 release? Will it also be available on the Microsoft store?

**Reply by aleksey-hoffman** [reply]
I think I'll finish v2 in under 3 months. 
I will upload it on Github in a few weeks as a separate branch and then, once it ready, I will replace the v1 with v2 on both Github and MS store

**Comment by MasihTak** [comment]
I would be happy to assist you if you need it with v2!

**Reply by aleksey-hoffman** [reply]
@MasihTak Hey, mate! I would love some help with v2. I will make the repo public on Github in a few days. Just need to clean it up a bit
Do you want to add me on Discord or Telegram and discuss it?

**Reply by MasihTak** [reply]
Cool! I made a Discord friend request!

**Comment by DanSketic** [comment]
Maybe the option to use a custom theme and icon set?

**Reply by aleksey-hoffman** [reply]
Hey, thanks for the suggestion. What do you mean by a custom theme? Something like cusomizable primary color?

**Reply by DanSketic** [reply]
Yes, absolutely. After the light and dark theme, the main colors could be set.  in the case of icons, follow this or add a colored icons.

**Comment by bassepeder** [comment]
When will the source code of version 2.0 be released?

**Reply by aleksey-hoffman** [reply]
@bastianpedersen about 2 more days, just need to configure the build file and remove debug code.

Thanks to the fresh tech stack, everyone will now be able to easily get in and tinker with it

**Reply by bassepeder** [reply]
I've been waiting for that moment ever since I bought myself an M1 Macbook! 😄

**Reply by aleksey-hoffman** [reply]
@bastianpedersen so you are a potential MacOS tester then 😄

Just keep in mind v2 is not finished yet, only the base is ready, now we'll need to port all the features with extensibility in mind (think of VSCode as a reference). It will have a tiny base, and all the advanced features will be added as plugins

**Reply by bassepeder** [reply]
I'll be looking into contributing. My Rust knowledge is limited, but the only way to improve that is to actually start using it in practice.

**Reply by bassepeder** [reply]
@aleksey-hoffman Any more updates?

**Reply by aleksey-hoffman** [reply]
@bastianpedersen hey, mate!
Yep, I uploaded the v2 version to [v2 branch](https://github.com/aleksey-hoffman/sigma-file-manager/tree/v2). It's in early development, it doesn't have most of the features yet. Only the base is ready, but everyone can now easily join in and help me build it up

**Reply by bassepeder** [reply]
I asked _before_ you uploaded the branch. Looks like you just did it 👍🏻

**Comment by bassepeder** [comment]
I am happy to report that Sigma is running on my M1.

**Reply by aleksey-hoffman** [reply]
@bastianpedersen 
That's great! Thanks for testing

Now when the difficult part is done, it's time to rewrite and migrate features and other systems (like user settings) from v1.

I'm also getting ready to publish the plugins system and make Sigma v2 truly extensible

**Reply by bassepeder** [reply]
I will look into contributing to this journey. I'm familiar with Vue. My Rust knowledge is limited, but I believe that putting that knowledge into practice will only speed up that process becoming better.

**Reply by aleksey-hoffman** [reply]
That's good thinking! Practice is what levels your skills up the most
 
If you decide to contribute, please make sure to first open a new issue, so we could discuss the feature and find the best approach

**Comment by hzxjy1** [comment]
Tauri will release v2.0 soon and will able to run webpage apps in mobile. Do you have any plans to migrate this project to the android platform?
